.github/workflows/prepare-release.yml

```diff
@@ -0,0 +1,55 @@
+name: Prepare release
+on:
+  workflow_dispatch:
+    inputs:
+      release_version:
+        description: new release version
+        required: true
+        default: (for example, 0.10.5)
+
+jobs:
+  release:
+    name: Prepare release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2.3.4
+
+      - name: Check input
+        run: |
+          if [[ ! "${{ github.event.inputs.release_version }}" =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            echo 'version "${{ github.event.inputs.release_version }}" not in ###.###.### format'
+            exit 1
+          fi
+          if [[ $( git ls-remote --tags origin refs/tags/v${{ github.event.inputs.release_version }} ) ]]; then
```

Comment:
What about an existing branch with the same name?

Reply:
It's possible and perfectly OK to have the same name for a branch and a tag, although anyone should really avoid it. The most critical thing is that we create a tag on the correct commit, which is what this workflow does, because we will build an artifact from a commit pointed to by the tag. So I'd say it is important to make sure that there is no existing tag with the same name that was likely created by a previous run of this workflow, e.g., due a build failure, or by that the person who triggered this workflow entered a wrong version. But for a branch name, it doesn't really matter. I think it isn't really worth checking branch names.

Reply:
I think it's a matter of error message. If the branch with the same name exists, line 31 fails with an appropriate error message.

```
git checkout -b release-v${{ github.event.inputs.release_version }}
```